cdap-watchdog/src/main/java/co/cask/cdap/logging/plugins/RollingLocationLogAppender.java

```diff
@@ -0,0 +1,253 @@
+                                        " but got " + context.getClass().getName());
+    }
+
+    started = true;
+  }
+
+  @Override
+  public void doAppend(ILoggingEvent eventObject) throws LogbackException {
+    try {
+      // logic from AppenderBase
+      if (!this.started) {
+        LOG.warn("Attempted to append to non started appender {}", this.getName());
+        return;
+      }
+
+      // logic from AppenderBase
+      if (getFilterChainDecision(eventObject) == FilterReply.DENY) {
+        return;
+      }
+
+      String namespaceId = eventObject.getMDCPropertyMap().get(LocationManager.TAG_NAMESPACE_ID);
+
+      if (namespaceId != null && !namespaceId.equals(NamespaceId.SYSTEM.getNamespace())) {
+        LocationIdentifier logLocationIdentifier = locationManager.getLocationIdentifier(eventObject
+                                                                                           .getMDCPropertyMap());
+        rollover(logLocationIdentifier, eventObject);
+        OutputStream locationOutputStream = locationManager.getLocationOutputStream(logLocationIdentifier, filePath);
+        setOutputStream(locationOutputStream);
+        writeOut(eventObject);
+      }
+    } catch (IllegalArgumentException iae) {
+      // this shouldn't happen
+      LOG.error("Unrecognized context ", iae);
+    } catch (IOException ioe) {
+      throw new LogbackException("Exception while appending event. ", ioe);
+    }
+  }
+
+  private void rollover(final LocationIdentifier logLocationIdentifier, ILoggingEvent logEvent) {
+    if (!locationManager.getActiveLocations().containsKey(logLocationIdentifier)) {
+      LOG.info("Decided not to roll over for first event");
+      return;
+    }
+    final LocationOutputStream locationOutputStream = locationManager.getActiveLocations().get(logLocationIdentifier);
+
+    if (triggeringPolicy instanceof LocationTriggeringPolicy) {
+      // no need to type cast on every event
+      if (locationTriggeringPolicy == null) {
+        locationTriggeringPolicy = ((LocationTriggeringPolicy) triggeringPolicy);
+      }
+
+      locationTriggeringPolicy.setLocation(locationOutputStream.getLocation());
+      // set number of bytes written to locationOutputStream, we need to do this because HDFS does not provide
+      // correct size of the file
+      locationTriggeringPolicy.setActiveLocationSize(locationOutputStream.getNumOfBytes());
+
+      if (locationTriggeringPolicy.isTriggeringEvent(logEvent)) {
+        try {
+          if (rollingPolicy instanceof LocationRollingPolicy) {
+            // no need to type cast on every event
+            if (locationRollingPolicy == null) {
+              locationRollingPolicy = ((LocationRollingPolicy) rollingPolicy);
+            }
+
+            locationRollingPolicy.setLocation(locationOutputStream.getLocation(), new Closeable() {
+              @Override
+              public void close() throws IOException {
+                locationOutputStream.close();
```

Comment:
I think we should not close it quietly because if there is any exception in closing the stream, we should know before renaming the file as part of roll over.